server/schedule/checker/rule_checker.go

```diff
@@ -222,11 +224,27 @@ func (c *RuleChecker) replaceUnexpectRulePeer(region *core.RegionInfo, rf *place
 		}
 	}
 
+	var witness *metapb.Peer
+	for _, witness = range region.GetWitnesses() {
+		if witness.StoreId != peer.StoreId && witness.Id != leaderID {
+			break
+		}
+	}
+
 	createOp := func() (*operator.Operator, error) {
 		if newLeader != nil && newLeader.GetId() != peer.GetId() {
 			return operator.CreateReplaceLeaderPeerOperator("replace-rule-"+status+"-leader-peer", c.cluster, region, operator.OpReplica, peer.StoreId, newPeer, newLeader)
 		}
-		return operator.CreateMovePeerOperator("replace-rule-"+status+"-peer", c.cluster, region, operator.OpReplica, peer.StoreId, newPeer)
+		/* TODO: enable this after the pr balance witness merged
+		if newPeer.IsWitness {
+			return operator.CreateMoveWitnessOperator("replace-rule-"+status+"-witness-peer", c.cluster, region, peer.StoreId, newPeer.StoreId, operator.OpWitness)
+		}
+		*/
+		if witness == nil {
+			return operator.CreateMovePeerOperator("replace-rule-"+status+"-peer", c.cluster, region, operator.OpReplica, peer.StoreId, newPeer)
+		}
+		newPeer.IsWitness = true
+		return operator.CreateMovePeerAndPromoteWitnessToVoterOperator("replace-rule-"+status+"-peer", c.cluster, region, operator.OpReplica, peer.StoreId, newPeer, witness.StoreId)
```

Comment:
I think it's a little far from the expectation because when we get here, it has already beyond the max store downtime. /cc  @Connor1996

Reply:
After discussing, remove logic for handling down peers to the next pr.